src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -285,11 +284,11 @@ void HMM<Distribution>::Train(const std::vector<arma::mat>& dataSeq,
     // emissions that are from this state.
     if (emissionList[state].size() > 0)
     {
-      arma::mat emissions(dimensionality, emissionList[state].size());
-      for (size_t i = 0; i < emissions.n_cols; i++)
+      std::vector<arma::vec> emissions(dimensionality);
+      for (size_t i = 0; i < emissionList[state].size(); i++)
       {
-        emissions.col(i) = dataSeq[emissionList[state][i].first].col(
-            emissionList[state][i].second);
+        emissions[i] = dataSeq[emissionList[state][i].first].col(
+        emissionList[state][i].second);
       }
 
       emission[state].Train(emissions);
```

Comment:
@rcurtin Why it keeps telling me no  **_matching member function for call to 'Train'_** But I have no idea why it would call the `Train` in `discrete distribution`. Could you give me some helps on that? Here is my full error I got: https://pastebin.mozilla.org/8971379